source/Halibut/Transport/ConnectionPoolAsync.cs

```diff
@@ -0,0 +1,188 @@
+using Halibut.Diagnostics;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Nito.AsyncEx;
+
+namespace Halibut.Transport
+{
+    public class ConnectionPoolAsync<TKey, TPooledResource> : IConnectionPool<TKey, TPooledResource> 
+        where TPooledResource : class, IPooledResource
+    {
+        readonly Dictionary<TKey, HashSet<TPooledResource>> pool = new();
+        readonly SemaphoreSlim poolLock = new(1, 1);
+
+        public int GetTotalConnectionCount()
```

Comment:
There is no async version of this as it's only used in tests

Reply:
Because we test this in a polymorphic way using `SyncOrAsync`, internal isn't possible.

🤔 I've been eyeing off this, so instead what I did is to just delete this method completely, and changed the test to test the behaviour instead of poking at an internal count 😁